core/objects_test.go

```diff
@@ -17,17 +17,25 @@ func TestRegistrationUpdate(t *testing.T) {
 	newURL, _ := ParseAcmeURL("http://new.invalid")
 	reg := Registration{
 		ID:        1,
-		Contact:   []*AcmeURL{oldURL},
+		Contact:   &[]*AcmeURL{oldURL},
 		Agreement: "",
 	}
 	update := Registration{
-		Contact:   []*AcmeURL{newURL},
+		Contact:   &[]*AcmeURL{newURL},
 		Agreement: "totally!",
 	}
 
 	reg.MergeUpdate(update)
-	test.Assert(t, len(reg.Contact) == 1 && reg.Contact[0] == update.Contact[0], "Contact was not updated %v != %v")
+	test.Assert(t, len(*reg.Contact) == 1 && (*reg.Contact)[0] == (*update.Contact)[0], "Contact was not updated %v != %v")
 	test.Assert(t, reg.Agreement == update.Agreement, "Agreement was not updated")
+
+	// Test that a registration contact can be removed by updating with an empty
```

Comment:
Let's break this into a separate test case, and test both the "remove contact" and "leave contact untouched" cases. Also, mostly in this test we construct native types, but it's probably good to parse from JSON for those particular cases, because the behavior depends on subtleties of our JSON parsing library.